mp/src/game/server/player_command.cpp

```diff
@@ -451,6 +494,29 @@ void CPlayerMove::RunCommand ( CBasePlayer *player, CUserCmd *ucmd, IMoveHelper
 		player->pl.v_angle.GetForModify() = player->GetLockViewanglesData();
 	}
 
+	// If the player is grounded, there is the possibility that they are bit above the ground and therefore might be
+	// above a trigger. The player could avoid this trigger by doing a jump, so to prevent this we extend the player
+	// collision by how much they are above the ground when checking for triggers
+	if (sv_ground_trigger_fix.GetBool() && player->GetGroundEntity() != nullptr)
+    {
+        Vector mins = player->CollisionProp()->OBBMins();
+        Vector maxs = player->CollisionProp()->OBBMaxs();
+        Vector offset(0.0f, 0.0f, sv_considered_on_ground.GetFloat());
+        
+        // CGameMovement::TryTouchGround
+        trace_t pm;
+        Ray_t ray;
+        ray.Init(player->GetAbsOrigin(), player->GetAbsOrigin() - offset, mins, maxs);
+        UTIL_TraceRay(ray, MASK_PLAYERSOLID, player, COLLISION_GROUP_PLAYER_MOVEMENT, &pm);
+        
+        if (pm.DidHit())
+        {
+            // Extend collision to ground (this will be undone in RunPostThink)
+            offset.z = player->GetAbsOrigin().z - pm.endpos.z;
+            player->SetCollisionBounds(mins - offset, maxs);
+        }
```

Comment:
Most of this logic is exactly the same as above (L444-469), you should pull out the common code into a small helper function.